redpanda/resources/serverlesscluster/data_serverless_cluster.go

```diff
@@ -76,19 +75,13 @@ func (d *DataSourceServerlessCluster) Read(ctx context.Context, req datasource.R
 		resp.Diagnostics.AddError(fmt.Sprintf("failed to read serverless cluster %s", model.ID), err.Error())
 		return
 	}
-	clusterURL, err := utils.SplitSchemeDefPort(serverlessCluster.DataplaneApi.Url, "443")
+	// Mapping the fields from the serverless cluster to the Terraform state
+	persist, err := generateModel(serverlessCluster)
 	if err != nil {
-		resp.Diagnostics.AddError("unable to parse Cluster API URL", err.Error())
+		resp.Diagnostics.AddError("unable to generate cluster model", err.Error())
```

Comment:
```suggestion
		resp.Diagnostics.AddError("unable to generate serverless cluster model", err.Error())
```

Reply:
Got rid of error entirely since the URL parsing was removed

Reply:
Even better!